app/Services/VersionService.php

```diff
@@ -138,12 +142,43 @@ public function createOrUpdateVersion(string $versionString, bool $allowed) : bo
      */
     public function toggleVersionAllowed(string $versionString)
     {
-        $version = Version::where('version', '=', $versionString);
+        $version = Version::withTrashed()->where('version', '=', $versionString)->first();
 
-        if (!$version->exists()) {
+        if (!$version) {
             throw new VersionNotFoundException('Version ' . $versionString . ' not found');
         }
 
-        $version->first()->toggleAllowed();
+        $version->toggleAllowed();
+    }
+
+    public function publishNewVersionFromGithub(string $tag)
+    {
+        if (Version::withTrashed()->where('version', $tag)->exists()) {
+            throw new VersionAlreadyExistsException();
+        }
+
+        // Create the version
+        Version::create(
+            [
+                'version' => $tag
+            ]
+        );
+
+        // Retire old versions
+        $versionsToKeep = Version::orderBy('id', 'desc')->limit(self::VERSIONS_TO_KEEP)->pluck('id')->toArray();
+        Version::whereNotIn('id', $versionsToKeep)->delete();
+    }
+
+    public function getFullVersionDetails(Version $version): array
+    {
+        $assetsUrl = "https://github.com/VATSIM-UK/uk-controller-plugin/releases/download/{$version->version}";
```

Comment:
Is the base URL worth putting as a config variable with a default for now? I get uneasy seeing hard coded URLs :P

Reply:
Done!